pandas/core/groupby/groupby.py

```diff
@@ -1754,44 +1754,48 @@ def _cython_agg_general(
         **kwargs,
     ):
         # Note: we never get here with how="ohlc" for DataFrameGroupBy;
-        #  that goes through SeriesGroupBy
+        # that goes through SeriesGroupBy
 
-        data = self._get_data_to_aggregate(numeric_only=numeric_only, name=how)
+        # Check to confirm numeric_only is fed either True or False and no other data type
+        if(isinstance(numeric_only, bool)):
+            data = self._get_data_to_aggregate(numeric_only=numeric_only, name=how)
```

Comment:
Can you change negate this condition and raise here; then the rest of this function can go untouched which makes the diff smaller. It also decreases the amount of indentation needed, improving readability.

Also use `is_bool` from `pandsa.core.dtypes.common`. We should accept e.g. `np.bool` here.